.circleci/config.yml

```diff
@@ -1,262 +1,147 @@
+          python-version: << parameters.python-version >>
+          cache: true
 
-    steps:
-      - checkout
-      - run:
-          name: build sdist
-          command: |
-            python -m venv env
-            . env/bin/activate
-            pip install -r requirements.txt
-            pip install build pip --upgrade
-            python -m build --sdist
-      - store_artifacts: *store-artifacts
-      - persist_to_workspace: *persist-to-workspace
-
-  deploy-all:
-    docker:
-      - image: cimg/python:3.12
-
-    steps:
-      - attach_workspace:
-          at: dist
+      - ocean/pip-install:
+          requirements: tests/requirements.txt
+          cache: false
 
-      - store_artifacts:
-          path: ./dist
+      - ocean/pip-install:
+          requirements: requirements.txt
+          packages: .
+          cache: false
 
-      - run:
-          name: deploy
-          command: |
-            python -m venv env
-            . env/bin/activate
-            python -m pip install twine
-            twine upload -u "$PYPI_USERNAME" -p "$PYPI_PASSWORD" --skip-existing ./dist/*
+      - ocean/coverage-run-unittest
 
-  test-codecov:
+  deploy:
     docker:
-      - image: cimg/python:3.12
+      - image: python:3.12
 
     steps:
       - checkout
+
       - run:
-          name: install dependencies
+          name: create virtualenv
           command: |
             python -m venv env
-            . env/bin/activate
-            pip install -r requirements.txt
-      - run: &unix-build
-          name: build
-          command: |
-            . env/bin/activate
-            pip install .
-      - run:
-          name: run coverage
-          command: |
-            . env/bin/activate
-            pytest tests/ --cov=dwave.plugins.torch
-      - codecov/upload
-
-  test-dependencies:
-    parameters:
-      python-version:
-        type: string
-      dependencies:
-        type: string
 
-    docker:
-      - image: python:<< parameters.python-version >>-slim
-
-    steps:
-      - checkout
-      - attach_workspace:
-          at: dist
       - run:
-          name: install
+          name: build sdist and bdist
           command: |
-            python -m venv env
             . env/bin/activate
-            pip install dwave-pytorch-plugin --no-index -f dist/ --force-reinstall --no-deps
-            pip install << parameters.dependencies >> --upgrade --only-binary=numpy
-      - run: &unix-run-tests
-          name: run tests
-          command: |
-            . env/bin/activate
-            cd tests/
-            pip install -r requirements.txt
-            python -m unittest
-
-  test-sdist:
-    docker:
-      - image: cimg/python:3.12
+            pip install -U pip setuptools wheel
+            python setup.py sdist
+            python setup.py bdist_wheel
 
-    steps:
-      - checkout
-      - attach_workspace:
-          at: dist
       - run:
-          name: install from sdist
+          name: upload
```

Comment:
you _could_ replace this whole step with
```
- ocean/pypi-deploy
```

but build and tests steps could also be otherwise simplified with `ocean` orb.. I'll update the orb, and create some nice examples we can use in the future.